matchzoo/models/bimpm.py

```diff
@@ -0,0 +1,420 @@
+"""An implementation of BiMPM Model."""
+import typing
+
+import torch
+import torch.nn as nn
+from torch.nn import functional as F
+
+from matchzoo.engine.param_table import ParamTable
+from matchzoo.engine.param import Param
+from matchzoo.engine.base_model import BaseModel
+from matchzoo.dataloader import callbacks
+
+
+class BiMPM(BaseModel):
+    """
+    BiMPM Model.
+
+    Reference:
+    - https://github.com/galsang/BIMPM-pytorch/blob/master/model/BIMPM.py
+
+    Examples:
+        >>> model = BiMPM()
+        >>> model.guess_and_fill_missing_params(verbose=0)
```

Comment:
Please add parameter configuration details, as the example would be a tutorial for this model.